src/components/lineup/delineate.ts

```diff
@@ -0,0 +1,56 @@
+import { groupBy } from 'lodash'
+import moment from 'moment'
+
+import { LineupItem } from './types'
+
+const NOW = moment()
+const START_OF_DAY = moment(NOW).startOf('day')
+const START_OF_YESTERDAY = moment(NOW).subtract(1, 'days').startOf('day')
+const START_OF_WEEK = moment(NOW).startOf('week')
+const START_OF_LAST_WEEK = moment(NOW).subtract(1, 'week').startOf('week')
+const START_OF_MONTH = moment(NOW).startOf('month')
+
+const Delineations = Object.freeze({
+  TODAY: 'today',
+  YESTERDAY: 'yesterday',
+  EARLIER_THIS_WEEK: 'earlier this week',
+  LAST_WEEK: 'last week',
+  EARLIER_THIS_MONTH: 'earlier this month'
+})
+
+const getGroup = timestamp => {
+  const time = moment(timestamp)
+  if (time > START_OF_DAY) {
+    return Delineations.TODAY
+  }
+  if (time > START_OF_YESTERDAY) {
+    return Delineations.YESTERDAY
+  }
+  if (time > START_OF_WEEK) {
+    return Delineations.EARLIER_THIS_WEEK
+  }
+  if (time > START_OF_LAST_WEEK) {
+    return Delineations.LAST_WEEK
+  }
+  if (time > START_OF_MONTH) {
+    return Delineations.EARLIER_THIS_MONTH
+  }
+
+  const startOfMonth = moment(time).startOf('month')
+  if (startOfMonth.year() === NOW.year()) {
+    return startOfMonth.format('MMMM')
+  }
+  return startOfMonth.format('MMMM YYYY')
+}
+
+export const delineateByTime = (entries: LineupItem[]) => {
+  return Object.entries(groupBy<LineupItem>(entries, getGroup)).map(
+    ([title, data], index) => ({
+      // For the first group,
+      // set the title as empty string to prevent
+      // Delineator from being displayed
+      title: index === 0 ? '' : title,
```

Comment:
could we filter the first one out? probably missing some context here :)

Reply:
SectionList expects an array of sections like so 
```ts
        [
          {
            title: '',
            data: items
          }
        ]
```

The logic here is that we still want the section to exist, we just don't want to show a section delineator. Ideally the `renderSectionHeader` prop on SessionList would be passed an index, but it isn't. So I think this is probably the best approach, but agree it's pretty hacky